base/math.jl

```diff
@@ -536,38 +536,56 @@ Stacktrace:
 [...]
 ```
 """
-hypot(x::Number, y::Number) = hypot(promote(x, y)...)
-function hypot(x::T, y::T) where T<:Number
-    ax = abs(x)
-    ay = abs(y)
-    if ax < ay
-        ax, ay = ay, ax
-    end
+function hypot(x::Number, y::Number)
+    #= Keep specialized two argument version, because it is faster
+    than generic varargs version below =#
+    ay, ax = minmax(abs(float(x)), abs(float(y)))
     if iszero(ax)
         r = ay / oneunit(ax)
     else
         r = ay / ax
     end
-
     rr = ax * sqrt(1 + r * r)
 
-    # Use type of rr to make sure that return type is the same for
-    # all branches
     if isnan(r)
-        isinf(ax) && return oftype(rr, Inf)
-        isinf(ay) && return oftype(rr, Inf)
-        return oftype(rr, r)
+        if isinf(x) || isinf(y)
+            return oftype(rr, Inf)
+        else
+            return oftype(rr, r)
+        end
     else
         return rr
-    end
+   	end
 end
-
 """
     hypot(x...)
 
 Compute the hypotenuse ``\\sqrt{\\sum x_i^2}`` avoiding overflow and underflow.
 """
-hypot(x::Number...) = sqrt(sum(abs2(y) for y in x))
+hypot(x::Number...) = _hypot(promote(x...))
+function _hypot(x::NTuple{N,<:Number}) where {N}
```

Comment:
Now that #30471 is merged, maybe we should try getting rid of the `promote`.  That is, just define
```
function hypot(x::Number...)
   isempty(x) && error("invalid hypot() call without arguments")
   s = sum(x -> abs2(float(x)), x)
    if isnan(s)
        return any(isinf, x) ? oftype(sqrt(s), Inf) : sqrt(s) # IEEE 754
    elseif isinf(s) || s ≤ _floatmin(s) * (N-1) # FIXME: revisit when #30387 is resolved
        # if overflow/underflow, try normalization
        ma = maximum(x -> abs(float(x)), x)
       ...
```

Reply:
Thanks!  I think that does it. Let's see what CI reports.